store/data_source.go

```diff
@@ -68,6 +68,9 @@ func (raw *dataSourceRaw) toDataSource() *api.DataSource {
 	}
 }
 
+// Data sources are used widely. We need to cache them to optimize query latency.
+var dataSourceCache = map[int][]*api.DataSource{}
```

Comment:
Should we use sync.map to avoid panic caused by concurrent read-write

Reply:
Or use RWMutex to protect this map?

Reply:
Done.